pkg/csi/cinder/nodeserver.go

```diff
@@ -67,22 +66,14 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 
 	ephemeralVolume := req.GetVolumeContext()["csi.storage.k8s.io/ephemeral"] == "true"
 	if ephemeralVolume {
-		// See https://github.com/kubernetes/cloud-provider-openstack/issues/1493
-		klog.Warningf("CSI inline ephemeral volumes support is deprecated in 1.24 release.")
-		return nodePublishEphemeral(req, ns)
+		// See https://github.com/kubernetes/cloud-provider-openstack/issues/2599
+		return nil, status.Error(codes.Unimplemented, "CSI inline ephemeral volumes support is removed in 1.31 release.")
 	}
 
 	// In case of ephemeral volume staging path not provided
 	if len(source) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "NodePublishVolume Staging Target Path must be provided")
 	}
-	_, err := ns.Cloud.GetVolume(volumeID)
-	if err != nil {
-		if cpoerrors.IsNotFound(err) {
-			return nil, status.Error(codes.NotFound, "Volume not found")
-		}
-		return nil, status.Errorf(codes.Internal, "GetVolume failed with error %v", err)
-	}
```

Comment:
Why do you remove this? This doesn't exactly feel like it's related, with `"csi.storage.k8s.io/ephemeral" == "true"` this code won't be exected.

Reply:
Hi, the main idea is - to remove openstack credentials dependency  from node csi daemonset.

We do not need to check the existence of the Volume in the daemonset by openstack api. When the CSI sidecar calls the NodeStageVolume/NodePublishVolume function, the Volume is already attached to the VM. We just need to find the block device in the /dev/* directory. If any issues occur, the block device mounter will return an error.

I believe, this check `_, err := ns.Cloud.GetVolume(volumeID)` is not necessary here.

What do you think?

Reply:
It's been a while ago when this check got in in https://github.com/kubernetes/cloud-provider-openstack/pull/676. @jichenjc, do you remember why it was necessary?

@kayrus, what do you think about this?

Reply:
seems last update is commit cc73e20cfe8233ff8ad3dbd040866f164225f6f8
but didn't find the commit message realted 

I am guessing this might prevent some concurrent deletion during the actions here ..